lib/stats_reporter.rb

```diff
@@ -0,0 +1,98 @@
+# generates text for a report on storage root size and object status
+class StatsReporter
+
+  require 'text-table'
+  require 'open3'
+
+  def self.storage_report_text
+    head = %w[filesystem total used pct_used free pct_free]
+    Text::Table.new(head: head, rows: storage_report_lines).to_s
+  end
+
+  def workflow_report_text
+    head = %w[workflow waiting error recent archived]
+    Text::Table.new(head: head, rows: workflow_report_lines).to_s
+  end
+
+  def self.workflow_xml
+    File.open('config/workflows/sdr/preservationIngestWF') { |f| Nokogiri::XML(f) }
+  end
+
+  def self.repository
+    workflow_xml.at_xpath('//workflow-def/@repository').value
+  end
+
+  def self.ingest_wf
+    workflow_xml.at_xpath('//workflow-def/@id').value
+  end
+
+  def self.ingest_wf_steps
+    workflow_xml.xpath('//process/@name').map(&:value)
+  end
+
+  def self.df_output
+    stdout_str, _status = Open3.capture2('df -h')
+    stdout_str
+  end
+
+  def self.storage_mount_lines
+    df_output.split("\n").select { |line| line.include?('services-disk') }
+  end
+
+  def self.parsed_storage_mount_lines
+    storage_mount_lines.map do |line|
+      storage_root = {}
+      split_line = line.split(' ')
```

Comment:
Unless I'm misreading the param, this is just restating `.split`'s default (split on whitespace).